tests/llmobs/conftest.py

```diff
@@ -212,3 +215,53 @@ def mock_ragas_evaluator(mock_llmobs_eval_metric_writer, ragas):
     LLMObsMockRagas.return_value = 1.0
     yield RagasFaithfulnessEvaluator
     patcher.stop()
+
+
+@pytest.fixture
+def tracer():
+    return DummyTracer()
+
+
+@pytest.fixture
+def llmobs_env():
+    return {
+        "DD_API_KEY": "<default-not-a-real-key>",
+        "DD_LLMOBS_ML_APP": "unnamed-ml-app",
+    }
+
+
+class TestLLMObsSpanWriter(LLMObsSpanWriter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.events = []
+
+    def enqueue(self, event):
+        self.events.append(event)
+
+
+@pytest.fixture
+def llmobs_span_writer():
+    yield TestLLMObsSpanWriter(interval=1.0, timeout=1.0)
+
+
+@pytest.fixture
+def llmobs(monkeypatch, tracer, llmobs_env, llmobs_span_writer):
+    for env, val in llmobs_env.items():
+        monkeypatch.setenv(env, val)
+
+    # Hack to reset the global config
+    ddtrace.llmobs._llmobs.config = Config()
+    ddtrace.llmobs._utils.config = Config()
```

Comment:
we can follow up here to have the llmobs components use a reference like `ddtrace.config._llm_ml_app` rather than keeping a local ref like `from ddtrace import config; config._llm_ml_app...` or we pass the config as a reference to our LLMObs instance.

Reply:
Yeah I'm a fan of either passing the config to our LLMObs instance, or even having our own LLMObs config to avoid being dependent on ddtrace global config settings.

Reply:
will follow up with a proposal for this!